lib/internal/cli_table.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const ansiRegex = () =>
+  new RegExp([
+    '[\\u001B\\u009B][[\\]()#;?]*(?:(?:(?:[a-zA-Z\\d]*',
+    '(?:;[a-zA-Z\\d]*)*)?\\u0007)',
+    '(?:(?:\\d{1,4}(?:;\\d{0,4})*)?[\\dA-PRZcf-ntqry=><~]))',
+  ].join('|'), 'g');
+
+const { Buffer } = require('buffer');
+
+const TableChars = {
+  /* eslint-disable node-core/non-ascii-character */
+  middle_middle: '─',
+  row_middle: '┼',
+  top_right: '┐',
+  top_left: '┌',
+  left_middle: '├',
+  top_middle: '┬',
+  bottom_right: '┘',
+  bottom_left: '└',
+  bottom_middle: '┴',
+  right_middle: '┤',
+  left: '│ ',
+  right: ' │',
+  middle: ' │ ',
+  /* eslint-enable node-core/non-ascii-character */
+};
+
+function countSymbols(string) {
+  const normalized = string.replace(ansiRegex(), '').normalize('NFC');
+  return Buffer.from(normalized, 'UCS-2').byteLength / 2;
+}
+
+const renderRow = (row, columnWidths) => {
+  let out = TableChars.left;
+  for (var i = 0; i < row.length; i++) {
+    const cell = row[i];
+    const len = countSymbols(cell);
+    const needed = (columnWidths[i] - len) / 2;
+    out += `${' '.repeat(needed)}${cell}${' '.repeat(Math.ceil(needed))}`;
+    if (i !== row.length - 1)
+      out += TableChars.middle;
+  }
+  out += TableChars.right;
+  return out;
+};
+
+const table = (head, columns) => {
+  const rows = [];
+  const columnWidths = head.map((h) => countSymbols(h));
+  const longestColumn = columns.reduce((n, a) => Math.max(n, a.length), 0);
+
+  for (var i = 0; i < head.length; i++) {
+    const column = columns[i];
+    for (var j = 0; j < longestColumn; j++) {
+      if (!rows[j])
+        rows[j] = [];
+      const v = rows[j][i] = column.hasOwnProperty(j) ? column[j] : '';
```

Comment:
I am not sure I understand the necessity for `hasOwnProperty` here. Can you elaborate?

Reply:
columns may not be filled all the way to the bottom, or may have empty slots, this replaces them with an empty string for rendering